attestation-agent/attestation-agent/src/bin/ttrpc-aa-client.rs

```diff
@@ -0,0 +1,151 @@
+    GetEvidence(GetEvidenceArgs),
+
+    /// Get attestation token
+    GetToken(GetTokenArgs),
+
+    /// Extend runtime measurement
+    ExtendRuntimeMeasurement(ExtendRuntimeMeasurementArgs),
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct GetEvidenceArgs {
+    /// base64 encodede runtime data
+    #[arg(short, long)]
+    runtime_data: String,
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct GetTokenArgs {
+    /// token type
+    #[arg(short, long)]
+    token_type: String,
+}
+
+#[derive(Args)]
+#[command(author, version, about, long_about = None)]
+struct ExtendRuntimeMeasurementArgs {
+    /// domain name
+    #[arg(short, long)]
+    domain: String,
+
+    /// operation name
+    #[arg(short, long)]
+    operation: String,
+
+    /// content name
+    #[arg(short, long)]
+    content: String,
+}
+
+#[tokio::main]
+pub async fn main() {
+    let args = Cli::parse();
+    let inner =
+        ttrpc::asynchronous::Client::connect(&args.attestation_sock).expect("connect ttrpc socket");
+    let client = AttestationAgentServiceClient::new(inner);
+    match args.operation {
+        Operation::GetTee => {
+            let req = GetTeeTypeRequest {
+                ..Default::default()
+            };
+            let res = client
+                .get_tee_type(context::with_timeout(TIMEOUT), &req)
+                .await
+                .expect("request to AA");
+            println!("{}", res.tee);
+        }
+        Operation::GetEvidence(args) => {
+            let runtime_data = base64::engine::general_purpose::STANDARD
+                .decode(args.runtime_data)
+                .unwrap();
+            let req = GetEvidenceRequest {
+                RuntimeData: runtime_data,
+                ..Default::default()
+            };
+            let res = client
+                .get_evidence(context::with_timeout(TIMEOUT), &req)
+                .await
+                .expect("request to AA");
+            let evidence = String::from_utf8(res.Evidence).unwrap();
+            println!("{evidence}");
+        }
+        Operation::GetToken(get_token_args) => {
+            let req = GetTokenRequest {
+                TokenType: get_token_args.token_type,
+                ..Default::default()
+            };
+            let res = client
+                .get_token(context::with_timeout(TIMEOUT), &req)
+                .await
+                .expect("request to AA");
+            let token = String::from_utf8(res.Token).unwrap();
+            println!("{token}");
+        }
+        Operation::ExtendRuntimeMeasurement(extend_runtime_measurement_args) => {
+            let req = ExtendRuntimeMeasurementRequest {
+                Domain: extend_runtime_measurement_args.domain,
+                Operation: extend_runtime_measurement_args.operation,
+                Content: extend_runtime_measurement_args.content,
+                ..Default::default()
```

Comment:
Can we add an (optional) arg for the register?

Reply:
Added.